modules/devices/x86/processor.c

```diff
@@ -448,14 +448,15 @@ gchar *caches_summary(GSList * processors)
     return ret;
 }
 
-#define PROC_SCAN_READ_BUFFER_SIZE 896
+#define PROC_SCAN_READ_BUFFER_SIZE 1024
 GSList *processor_scan(void)
 {
     GSList *procs = NULL, *l = NULL;
     Processor *processor = NULL;
     FILE *cpuinfo;
-    gchar buffer[PROC_SCAN_READ_BUFFER_SIZE];
+    gchar *buffer;
 
+    buffer = (gchar *) g_malloc(PROC_SCAN_READ_BUFFER_SIZE * sizeof(gchar));
```

Comment:
```suggestion
    buffer = g_malloc(PROC_SCAN_READ_BUFFER_SIZE);
```

(`sizeof(gchar)` is always going to be 1, and `gpointers` will always cast to any pointer automatically.)